File: src/main/java/fr/labs/sza/tournoiApplication.java
package fr.labs.sza;

import fr.labs.sza.resources.JoueurResource;
import fr.labs.sza.resources.TournoiResource;
import io.dropwizard.Application;
import io.dropwizard.setup.Bootstrap;
import io.dropwizard.setup.Environment;

public class tournoiApplication extends Application<tournoiConfiguration> {

    public static void main(final String[] args) throws Exception {
        new tournoiApplication().run(args);
    }

    @Override
    public String getName() {
        return "tournoiKotlin";
    }

    @Override
    public void initialize(final Bootstrap<tournoiConfiguration> bootstrap) {
        // TODO: application initialization
    }

    @Override
    public void run(final tournoiConfiguration configuration,
                    final Environment environment) {
        environment.jersey().register(new JoueurResource());
        environment.jersey().register(new TournoiResource());
    }

}

File: src/main/java/fr/labs/sza/tournoiApplication.kt
package fr.labs.sza

import fr.labs.sza.component.ApplicationComponent
import fr.labs.sza.component.DaggerApplicationComponent
import fr.labs.sza.resources.JoueurResource
import fr.labs.sza.resources.TournoiResource
import io.dropwizard.Application
import io.dropwizard.setup.Bootstrap
import io.dropwizard.setup.Environment

class tournoiApplication : Application<tournoiConfiguration>() {


    private var applicationComponent: ApplicationComponent? = null

    override fun getName(): String {
        return "tournoiKotlin"
    }

    override fun initialize(bootstrap: Bootstrap<tournoiConfiguration>?) {
        // TODO: application initialization
        applicationComponent = DaggerApplicationComponent.builder().build()
        applicationComponent!!.inject(this)
    }

    override fun run(configuration: tournoiConfiguration,
                     environment: Environment) {
        environment.jersey().register(JoueurResource())
        environment.jersey().register(TournoiResource())
    }

    companion object {

        @Throws(Exception::class)
        @JvmStatic
        fun main(args: Array<String>) {
            tournoiApplication().run(*args)
        }
    }

}
